protoc-gen-grpc-gateway/internal/gengateway/template.go

```diff
@@ -348,30 +352,15 @@ var (
 	{{$binding := .}}
 	{{range $param := .PathParams}}
 	{{$enum := $binding.LookupEnum $param}}
-	val, ok = pathParams[{{$param | printf "%q"}}]
-	if !ok {
-		return nil, metadata, status.Errorf(codes.InvalidArgument, "missing parameter %s", {{$param | printf "%q"}})
-	}
+
+	_, _ = val, ok
 {{if $param.IsNestedProto3}}
-	err = runtime.PopulateFieldFromPath(&protoReq, {{$param | printf "%q"}}, val)
-	if err != nil {
-		return nil, metadata, status.Errorf(codes.InvalidArgument, "type mismatch, parameter: %s, error: %v", {{$param | printf "%q"}}, err)
-	}
-	{{if $enum}}
-		e{{if $param.IsRepeated}}s{{end}}, err = {{$param.ConvertFuncExpr}}(val{{if $param.IsRepeated}}, {{$binding.Registry.GetRepeatedPathParamSeparator | printf "%c" | printf "%q"}}{{end}}, {{$enum.GoType $param.Method.Service.File.GoPkg.Path}}_value)
-		if err != nil {
-			return nil, metadata, status.Errorf(codes.InvalidArgument, "could not parse path as enum value, parameter: %s, error: %v", {{$param | printf "%q"}}, err)
-		}
-	{{end}}
+	_, _ = val, ok
 {{else if $enum}}
-	e{{if $param.IsRepeated}}s{{end}}, err = {{$param.ConvertFuncExpr}}(val{{if $param.IsRepeated}}, {{$binding.Registry.GetRepeatedPathParamSeparator | printf "%c" | printf "%q"}}{{end}}, {{$enum.GoType $param.Method.Service.File.GoPkg.Path}}_value)
-	if err != nil {
-		return nil, metadata, status.Errorf(codes.InvalidArgument, "type mismatch, parameter: %s, error: %v", {{$param | printf "%q"}}, err)
-	}
+	_, _ = val, ok
 {{else}}
-	{{$param.AssignableExpr "protoReq"}}, err = {{$param.ConvertFuncExpr}}(val{{if $param.IsRepeated}}, {{$binding.Registry.GetRepeatedPathParamSeparator | printf "%c" | printf "%q"}}{{end}})
-	if err != nil {
-		return nil, metadata, status.Errorf(codes.InvalidArgument, "type mismatch, parameter: %s, error: %v", {{$param | printf "%q"}}, err)
+	if err := runtime.PopulatePathParameters(&protoReq, pathParams, &utilities.DoubleArray{}); err != nil {
```

Comment:
Need help here. How much of this is still needed? The go-template is pretty complex so would appreciate any guidance on properly adding this branch.

Reply:
I would assume this is all still going to be required, I think you'll need an understanding of what each case does before we can consider replacing it. Remember that we want to have two way backward compatibility as much as possible, a user should be able to upgrade either the runtime or the generator without things breaking, ideally. If we're adding a new `runtime.ParsePathParameters` then it means new generated files can't keep using old versions of the library, which isn't great, but may be unavoidable for a feature like this.

The way to probably make this work is to copy whatever logic we have for filtering query parameters into `*utilities.DoubleArray` and reuse it for figuring out what parameters are path parameters, then use that as a parameter to `ParsePathParameters`, much like we do for query parameters today.

Reply:
Got it. Can you provide some hints on what the cases here mean? Theres a lot of general terms like `IsNestedProto3` used in the other if statements.

Reply:
I'm frankly not all that familiar with the deeper workings of this template either - I'd recommend diving into where it's [being instantiated and executed](https://github.com/grpc-ecosystem/grpc-gateway/blob/74ecd1deffacf97bcbee90e81c631ef6a3c275f2/protoc-gen-grpc-gateway/internal/gengateway/template.go#L151-L214) to understand the parameters involved. I think a useful initial test would be to just add a flag that can replace the path parameter parsing with a call to runtime.ParsePathParameter.

https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-grpc-gateway/internal/gengateway/template.go#L131-L143 is where the query parameter doubleArray is created (I don't know how that works tbh). We'll need somethin similar for paths.

Reply:
Got it. I'm not sure if we need the filter so I've removed it. Studying all the code branches now.